runtime/vm/jvminit.c

```diff
@@ -7896,3 +7910,119 @@ parseGlrOption(J9JavaVM* jvm, char* option)
 
 	return JNI_ERR;
 }
+
+static UDATA
+ensureHashedHashFn(void *entry, void *userData)
+{
+	J9UTF8 *hashtableEntry = *(J9UTF8 **)entry;
+
+	return computeHashForUTF8(J9UTF8_DATA(hashtableEntry), J9UTF8_LENGTH(hashtableEntry));
+}
+
+static UDATA
+ensureHashedHashEqualFn(void *lhsEntry, void *rhsEntry, void *userData)
+{
+	J9UTF8 *lhsHashtableEntry = *(J9UTF8 **)lhsEntry;
+	J9UTF8 *rhsHashtableEntry = *(J9UTF8 **)rhsEntry;
+
+	return J9UTF8_DATA_EQUALS(J9UTF8_DATA(lhsHashtableEntry), J9UTF8_LENGTH(lhsHashtableEntry), J9UTF8_DATA(rhsHashtableEntry), J9UTF8_LENGTH(rhsHashtableEntry));
+}
+
+static UDATA
+ensureHashedDoDelete(void *entry, void *userData)
+{
+	PORT_ACCESS_FROM_PORT((J9PortLibrary *)userData);
+
+	j9mem_free_memory(*(J9UTF8 **)entry);
+	return TRUE;
+}
+
+static void
+cleanupEnsureHashedConfig(J9JavaVM* jvm)
+{
+	PORT_ACCESS_FROM_JAVAVM(jvm);
+
+	if (NULL != jvm->ensureHashedClasses) {
+		hashTableForEachDo(jvm->ensureHashedClasses, ensureHashedDoDelete, PORTLIB);
+		hashTableFree(jvm->ensureHashedClasses);
+	}
+	jvm->ensureHashedClasses = NULL;
+}
+
+static UDATA
+parseEnsureHashedOption(J9JavaVM *jvm, char *option)
+{
+	PORT_ACCESS_FROM_JAVAVM(jvm);
+
+	if (0 == strcmp(option, "")) {
+		/* Empty class name */
```

Comment:
You may need an error message here. Look for J9NLS_VM_XXX in this file as examples.